src/Symfony/Component/Workflow/Dumper/PlantUmlDumper.php

```diff
@@ -0,0 +1,199 @@
+ */
+class PlantUmlDumper implements DumperInterface
+{
+    const INITIAL = 'initial';
+
+    const MARKED = 'marked';
+
+    const SYMFONY_LOGO = 'sprite $sf_logo [81x20/16z] {
+hPNRaYiX24K1xwBo_tyx6-qaCtDEJ-KXLYMTLbp0HWcHZr3KRDJ8z94HG3jZn4_mijbQ2ryJoFePtXLWA_qxyGy19DpdY_10z11ZAbGjFHRwcEbcKx5-wqsV
+yIMo8StMCHKh8ZUxnEwrZiwRAUOvy1lLcPQF4lEFAjhzMd5WOAqvKflS0Enx8PbihiSYXM8ClGVAseIWTAjCgVSAcnYbQG79xKFsZ0VnDCNc7AVBoPSMcTsX
+UnrujbYjjz0NnsObkTgnmolqJD4QgGUYTQiNe8eIjtx4b6Vv8nPGpncn3NJ8Geo9W9VW2wGACm_JzgIO8A8KXr2jUBCVGEAAJSZ6JUlsNnmOzmIYti9G7bjL
+8InaHM9G40NkwTG7OxrggvNIejA8AZuqyWjOzTIKi-wwYvjeHYesSWuPiTGDN5THzkYLU4MD5r2_0PDhG7LIUG33z5HtM6CP3icyWEVOS61sD_2ZsBfJdbVA
+qM53XHDUwhY0TAwPug3OG9NonRFhO8ynF3I4unuAMDHmSrXH57V1RGvl9jafuZF9ZhqjWOEh98y0tUYGsUxkBSllIyBdT2oM5Fn2-ut-fzsq_cQNuL6Uvwqr
+knh4RrvOKzxZfLV3s0rs_R_1SdYt3VxeQ1_y2_W2
+}';
+
+    protected static $defaultOptions = array(
+        'skinparams' => array(
+            'titleBorderRoundCorner' => 15,
+            'titleBorderThickness' => 2,
+            'state' => array(
+                'BackgroundColor<<initial>>' => '#87b741',
+                'BackgroundColor<<marked>>' => '#3887C6',
+                'BorderColor' => '#3887C6',
+                'BorderColor<<marked>>' => 'Black',
+                'FontColor<<marked>>' => 'White',
+            ),
+            'agent' => array(
+                'BackgroundColor' => '#ffffff',
+                'BorderColor' => '#3887C6',
+            ),
+        ),
+    );
+
+    const DEFAULT_TRANSITION_TYPE = 'arrow';
+
+    const SQUARE_TRANSITION_TYPE = 'square';
+
+    protected $transitionType;
+
+    public function __construct($transitionType)
+    {
+        $this->transitionType = static::DEFAULT_TRANSITION_TYPE;
+        if ($transitionType == static::SQUARE_TRANSITION_TYPE) {
+            $this->transitionType = $transitionType;
+        }
+    }
+
+    protected function isSquareTransitionType()
+    {
+        return $this->transitionType == static::SQUARE_TRANSITION_TYPE;
+    }
+
+    /**
+     * @param Definition   $definition
+     * @param Marking|null $marking
+     * @param array        $options
+     */
+    public function dump(Definition $definition, Marking $marking = null, array $options = array())
+    {
+        $options = array_replace_recursive(self::$defaultOptions, $options);
+        $code = $this->initialize($options);
+        foreach ($definition->getPlaces() as $place) {
+            $code[] =
+                "state {$place}".
+                ($definition->getInitialPlace() === $place ? ' <<'.self::INITIAL.'>>' : '').
+                ($marking && $marking->has($place) ? ' <<'.self::MARKED.'>>' : '');
+        }
+        if ($this->isSquareTransitionType()) {
+            foreach ($definition->getTransitions() as $transition) {
+                $code[] = "agent {$transition->getName()}";
+            }
+        }
+        foreach ($definition->getTransitions() as $transition) {
```

Comment:
It feels like alot of complexity comparing to the GraphvizDumper, isn't in the class something you could re-use to get arrays and then transform them directly ?

Reply:
Not really sure I get what you want me to do. I agree it is not really straightforward but it is an efficient way to manage both TransitionType, GraphvizDumper as 2 classes to do that. I did not see the value for the PlantUML one, and I still don't see it. (but I would love to have your opinion)

I could do that in another method and then merge arrays... is it what you would like?.

Let me know!

Thank you!

Reply:
I was just thinking on how much this adds complexity. I guess if merging the arrays cost more in performance than what you are doing here let's keep this. (I think it will be anyway).

I just wanted to know more about this ;).

Reply:
ok